pkg/operators/kubemanager/kubemanager.go

```diff
@@ -337,6 +343,166 @@ func (m *KubeManagerInstance) EnrichEvent(ev any) error {
 	return nil
 }
 
+func (k *KubeManager) GlobalParams() api.Params {
+	return apihelpers.ParamDescsToParams(k.GlobalParamDescs())
+}
+
+func (k *KubeManager) InstanceParams() api.Params {
+	return apihelpers.ParamDescsToParams(k.ParamDescs())
+}
+
+func (k *KubeManager) InstantiateDataOperator(ctx operators.GadgetContext, paramValues api.ParamValues) (
+	operators.DataOperatorInstance, error,
+) {
+	params := k.ParamDescs().ToParams()
+	err := params.CopyFromMap(paramValues, "")
+	if err != nil {
+		return nil, err
+	}
+
+	traceInstance := &KubeManagerInstance{
+		manager:            k,
+		enrichEvents:       false,
+		attachedContainers: make(map[string]*containercollection.Container),
+		params:             params,
+		gadgetCtx:          ctx,
+		id:                 uuid.New().String(),
+
+		eventWrappers: make(map[datasource.DataSource]*compat.EventWrapperBase),
+	}
+
+	// hack - this makes it possible to use the Attacher interface
+	var ok bool
+	traceInstance.gadgetInstance, ok = ctx.GetVar("ebpfInstance")
+	if !ok {
+		return nil, fmt.Errorf("getting ebpfInstance")
+	}
+
+	activate := false
+
+	// Check, whether the gadget requested a map from us
+	if t, ok := ctx.GetVar(gadgets.MntNsFilterMapName); ok {
+		if _, ok := t.(*ebpf.Map); ok {
+			ctx.Logger().Debugf("gadget requested map %s", gadgets.MntNsFilterMapName)
+			activate = true
+		}
+	}
+
+	// Check for data sources containing refererences to mntns/netns that we could enrich data for
+	for _, ds := range ctx.GetDataSources() {
+		mntnsFields := ds.GetFieldsWithTag("type:gadget_mntns_id")               // TODO: const
+		netnsFields := ds.GetFieldsWithTag("type:gadget_netns_id", "name:netns") // TODO: const
+		if len(mntnsFields) == 0 && len(netnsFields) == 0 {
+			continue
+		}
+
+		ctx.Logger().Debugf("ds found: %q", ds.Name())
+
+		activate = true
+
+		var err error
+
+		var mntnsField datasource.FieldAccessor
+		var netnsField datasource.FieldAccessor
+
+		for _, f := range mntnsFields {
+			ctx.Logger().Debugf("using mntns enrichment")
+			mntnsField = f
+			mntnsField.SetHidden(true, false)
```

Comment:
I'll change it, but I'm still worried about "special" ways for _any_ field a bit.

IMHO, even if _we_ think, we'd like to have something handled in another way and it isn't 100% applicable to _all_ situations, we ourselves should also use metadata additions instead of hardcoding a certain behavior (which is the case for imageName+imageDigest). But let's discuss these things later on as you proposed in another comment :)